Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM debian:buster-slim
+
+# Common tools
+RUN apt update && apt install -y ca-certificates curl gnupg make zip unzip wget apt-utils
+RUN mkdir -p /etc/apt/keyrings && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+
+# Nodejs
+ARG NODE_MAJOR
+RUN echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_${NODE_MAJOR}.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list
+RUN apt update
+RUN apt install -y nodejs --no-install-recommends
+
+# Python
+RUN apt install -y build-essential zlib1g-dev libffi-dev libssl-dev libgdbm-dev libc6-dev libbz2-dev --no-install-recommends
```

Comment:
I would probably combine all the apt installs into one command. In docker every line is a layer and the more layers you have, the larger the image is and the slower the build time is. If we put all the install steps in one layer, i.e one RUN command, it will cache and do it first and given it doesn't change much, it will have no impact.